bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt

```diff
@@ -157,80 +157,125 @@ internal fun newErrorMetricsEvent(
   error: BKTException,
   apiId: ApiId,
 ): Event {
-  val metricEventType = error.toMetricEventType()
   return Event(
     id = idGenerator.newId(),
     type = EventType.METRICS,
-    event = EventData.MetricsEvent(
-      timestamp = clock.currentTimeSeconds(),
-      type = metricEventType,
-      event = newErrorMetricsEventData(featureTag, apiId, metricEventType),
-      sdkVersion = BuildConfig.SDK_VERSION,
-      metadata = newMetadata(appVersion),
-    ),
+    event = error.toEventDataMetricEvent(clock.currentTimeSeconds(), featureTag, appVersion, apiId),
   )
 }
-internal fun newErrorMetricsEventData(
+
+internal fun BKTException.toEventDataMetricEvent(
+  timestamp: Long,
   featureTag: String,
+  appVersion: String,
   apiId: ApiId,
-  type: MetricsEventType,
-): MetricsEventData {
+): EventData.MetricsEvent {
   val labels = mapOf("tag" to featureTag)
+  val typeAndEvent: Pair<MetricsEventType, MetricsEventData> =
```

Comment:
```suggestion
  val (type, event) = when (this) {
```

Reply:
updated in https://github.com/bucketeer-io/android-client-sdk/pull/91/commits/b30b75f1cb80136aed2d4ec70258a19f3d47d9d0